packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -584,8 +584,13 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
   }
 
   @override
-  void applyPaintTransform(RenderObject child, Matrix4 transform) {
-    applyPaintTransformForBoxChild(child as RenderBox, transform);
+  void applyPaintTransform(RenderBox child, Matrix4 transform) {
+    if (_keepAliveBucket.containsKey(indexOf(child))) {
+      // The children that are kept alive can't be drawn.
```

Comment:
This looks OK to me although it would help if the comment explained how we end up applying a paint transform to a child that's not laid out or painted.

FWIW, I noticed that RenderFittedBox's applyPaintTransform() also uses transform.setZero() to defeat painting a child.